common/junit-platform-native/src/main/java/org/graalvm/junit/platform/UniqueIdTrackingTestExecutionListener.java

```diff
@@ -71,8 +71,17 @@
 
     private final List<String> uniqueIds = new ArrayList<>();
 
+    @Override
+    public void executionSkipped(TestIdentifier testIdentifier, String reason) {
+        addTest(testIdentifier);
```

Comment:
We also need to include tests that were skipped on the JVM in the native-image used for testing. If not, we would not be able to run tests disabled through `DisableIf` and `EnableIf` on the native-image, even if the condition that disabled them changed in the meantime.

Reply:
That's an interesting topic. On the one hand, I can see that people might want to have each JUnit Jupiter `ExecutionCondition` reevaluated within the native image. On the other hand, people might want literally the exact same tests executed without having to have the conditions reevaluated (maybe b/c the conditions don't work the same within the native image).

Though perhaps it's best to include them by default now. We could later add an option to make that behavior configurable.

BUT... if we make this change here, I should probably make the same change in JUnit 5.8's `UniqueIdTrackingListener`.

Reply:
Both requirements seem valid - in this case, it may be best to go with the current approach and include an option later that allows the users to skip condition evaluation on the native-image. Note that this is a bit tricky: we are still evaluating the conditions, but we're only doing it for the tests whose condition allowed their execution on the JVM - this means that the set of tests we could execute on native before this change was either the same or a subset of the tests that had ran on the JVM

Reply:
I agree with changing the `UniqueIdTracingListener` - initially, it could also include both executed and skipped tests. If we move forward with adding an option to skip tests that were skipped on the JVM, we might need to update the output file format to also provide test execution information for each tests (by information, I mean either executed or skipped, not succeeded or failed)

Reply:
> If we move forward with adding an option to skip tests that were skipped on the JVM, we might need to update the output file format to also provide test execution information for each tests (by information, I mean either executed or skipped, not succeeded or failed)

Why would we need to differentiate between those?

If a test method was disabled during the JVM run, it may or may not be disabled within the native image run. In either case, we would still select it by it's unique ID, and the JUnit Jupiter `ExecutionCondition` would evaluate it within the native image.

Reply:
Ah - that was related to the above proposal for an option that allows people to skip tests that were skipped on the JVM. That way, we could differentiate whether the test was skipped or executed from inside the feature. However, I agree with you - it probably isn't needed for now